src/extra/others/fragment/lv_fragment.c

```diff
@@ -0,0 +1,392 @@
+    manager_stack_t * item = item_new(cls);
+    lv_obj_t * parent = manager->container;
+    item_create_fragment(manager, item, args);
+    /* Destroy object of previous screen */
+    if(manager->top) {
+        item_destroy_obj(manager, manager->top);
+    }
+    item_create_obj(manager, item, parent, NULL);
+    manager_stack_t * top = manager->top;
+    item->prev = top;
+    manager->top = item;
+}
+
+void lv_fragment_manager_replace(lv_fragment_manager_t * manager, const lv_fragment_class_t * cls,
+                                 void * args)
+{
+    LV_ASSERT(manager);
+    LV_ASSERT(cls);
+    manager_stack_t * top = item_new(cls);
+    item_create_fragment(manager, top, args);
+    manager_stack_t * old = manager->top;
+    if(old) {
+        item_destroy_obj(manager, old);
+        item_destroy_fragment(old);
+        lv_mem_free(old);
+    }
+    manager->top = top;
+    item_create_obj(manager, top, manager->container, NULL);
+}
+
+size_t lv_fragment_manager_size(lv_fragment_manager_t * manager)
+{
+    size_t size = 0;
+    for(manager_stack_t * cur = manager->top; cur; cur = cur->prev) {
+        size++;
+    }
+    return size;
+}
+
+void lv_fragment_manager_pop(lv_fragment_manager_t * manager)
+{
+    LV_ASSERT(manager);
+    manager_stack_t * top = manager->top;
+    if(!top) return;
+    manager_stack_t * prev = top->prev;
+    bool dialog = top->dialog;
+    if(!dialog && prev) {
+        item_create_fragment(manager, prev, NULL);
+    }
+    item_destroy_obj(manager, top);
+    item_destroy_fragment(top);
+    lv_mem_free(top);
+    if(!dialog && prev) {
+        item_create_obj(manager, prev, manager->container, NULL);
+    }
+    manager->top = prev;
+}
+
+bool lv_fragment_manager_dispatch_event(lv_fragment_manager_t * manager, int which, void * data1, void * data2)
+{
+    LV_ASSERT(manager);
+    manager_stack_t * top = manager->top;
+    if(!top) return false;
+    lv_fragment_t * instance = top->instance;
+    if(!instance || !top->cls->event_cb) return false;
+    return top->cls->event_cb(instance, which, data1, data2);
+}
+
+#if LV_USE_MSGBOX
+
+void lv_fragment_manager_show(lv_fragment_manager_t * manager, const lv_fragment_class_t * cls, void * args)
+{
+    LV_ASSERT(manager);
+    LV_ASSERT(cls);
+    manager_stack_t * item = item_new(cls);
+    item_create_fragment(manager, item, args);
+    item_create_obj(manager, item, NULL, &lv_msgbox_class);
+    item->dialog = true;
+    manager_stack_t * top = manager->top;
+    item->prev = top;
+    manager->top = item;
+}
+
+#endif
+
+lv_fragment_t * lv_fragment_manager_get_top(lv_fragment_manager_t * manager)
+{
+    LV_ASSERT(manager);
+    manager_stack_t * top = manager->top;
+    if(!top)return NULL;
+    return top->instance;
+}
+
+lv_fragment_t * lv_fragment_manager_get_parent(lv_fragment_manager_t * manager)
+{
+    LV_ASSERT(manager);
+    return manager->parent;
+}
+
+lv_fragment_t * lv_fragment_class_create_unmanaged(const lv_fragment_class_t * cls,
```

Comment:
What this slight change in the API:

Instead of:
```c
    lv_fragment_manager_t *manager = lv_fragment_manager_create(root, NULL);
    lv_fragment_manager_replace(manager, &sample_cls, "Fragment");
```

This:
```c
    lv_fragment_manager_t *manager = lv_fragment_manager_create(root, NULL);
    lv_fragment_t * f1 = lv_fragment_create(manager, &sample_cls, "Fragment");
    lv_fragment_manager_replace(manager, f1);
```
For unmanaged:
```c 
  lv_fragment_t * f1 = lv_fragment_create(NULL, &sample_cls, "Fragment");
```

It's a little bit more verbose but cleaner IMO.

Reply:
This makes it more like Android Fragment which really makes sense. But in this way I think I will have to separate fragment creation and object attachment.

Reply:
`object attachment` means creating object?

Reply:
It means creating `lv_obj_t`s upon containers. It's separated now.